src/semeio/fmudesign/_excel2dict.py

```diff
@@ -297,63 +297,66 @@ def _excel2dict_onebyone(
 
     designinput["sensname"] = designinput["sensname"].ffill()
 
-    # Read dependencies
     if "dependencies" in designinput:
+        valid_deps = designinput[designinput["dependencies"].notna()]
         inputdict["dependencies"] = OrderedDict()
-        for row in designinput.itertuples():
-            if _has_value(row.dependencies):
-                inputdict["dependencies"][row.param_name] = _read_dependencies(
-                    input_filename,
-                    row.dependencies,
-                    row.param_name,
-                )
+        for row in valid_deps.itertuples():
+            inputdict["dependencies"][row.param_name] = _read_dependencies(
+                input_filename,
+                str(row.dependencies),
+                str(row.param_name),
+            )
 
-    # Read decimals
     if "decimals" in designinput:
-        inputdict["decimals"] = OrderedDict()
-        for row in designinput.itertuples():
-            if _has_value(row.decimals) and _is_int(row.decimals):
-                inputdict["decimals"][row.param_name] = int(row.decimals)
+        # Convert to numeric, then filter for integers
+        numeric_decimals = pd.to_numeric(designinput["decimals"], errors="coerce")
```

Comment:
This decimals feature has only one integration test, thus not properly unit tested. It is more dangerous to refactor, can you add some unit tests for the feature that works before and after?

Reply:
`decimals` column is specified in `design_input_mc_with_correls.xlsx` which is run by for example `test_generate_full_mc_snapshot` so it should catch any changes. Many things are lacking unit-tests in `fmudesign`.